src/main/java/com/scaleunlimited/flinkcrawler/tools/CrawlTool.java

```diff
@@ -257,19 +294,26 @@ public static void main(String[] args) {
 	}
 
 	public static void run(StreamExecutionEnvironment env, CrawlToolOptions options) throws Exception {
-		UserAgent userAgent = new UserAgent("flink-crawler", "flink-crawler@scaleunlimited.com", "https://github.com/ScaleUnlimited/flink-crawler/wiki/Crawler-Policy");
 		
-		SimpleUrlValidator urlValidator =
+	    // TODO Complain if -cachedir is specified when not running locally?
+	    
+	    SimpleUrlValidator urlValidator =
 			(	options.isSingleDomain() ?
 				new SingleDomainUrlValidator(options.getSingleDomain())
 			:	new SimpleUrlValidator());
 		
+		UserAgent userAgent =
+	        (   options.isCommonCrawl() ?
+                new UserAgent("unused-common-crawl-user-agent", "", "")
+            :   options.getUserAgent());
```

Comment:
So the post-parsing check would want to ensure that if it's not a common crawl, then the user agent is required (but the Args4J setting has to be optional)

Reply:
Again, I don't think we need a post-parsing check.

Reply:
Both the user agent and common crawl parameters have to be optional, yes? But if neither is set, then that's an error, right? So why wouldn't you need a post-parsing check?

Reply:
There is a check later if neither is specified, yes, but that's in CrawlTopology.

Reply:
OK, so there is a post-parsing check...I think it would be better to catch invalid options higher up, e.g. where we'd print out some standard message about invalid options.

Reply:
I added a validate() method that also gets called automatically from all the get methods.